### PR TITLE
doc: Event manager

### PR DESCRIPTION
**Pull request by emob-nordic**
Update event manager doc after CAF is introduced.

Jira: NCSDK-8410

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

### CODE REVIEW

File: include/event_manager.rst [hunk 1]
```diff
@@ -186,15 +137,74 @@ The following code example shows a source file for the event type ``sample_event
...
+Modules
+*******
+
+Modules are separated source file that can subscribe to every defined event.
+Each module can subscribe to multiple events, there is no any limitation.
+An application can have as many modules as required.
+
+Event manager provide API for subscribing modules to a specific events defined in the application.
+If a module is subscribing to a specific event, its called listener.
+Every listener is identified by a unique name.
```
Comment (greg-fer) [inline include/event_manager.rst:198]: I think these two lines should be in line 52, and the line from the line 52 should go here.

File: include/event_manager.rst [hunk 2]
```diff
@@ -260,18 +270,29 @@ The following code example shows how to register an event listener with an event
...
 	EVENT_SUBSCRIBE(sample_module, sample_event);
 
-All the events, both the ones with and the ones without the variable size data, are handled by an application module in the same way.
 The variable size data is accessed in the same way as the other members of the structure defining an event.
 
 
+Event Manager extensions
+************************
+
+Event Manager embeds additional features that could be helpful when debugging event-based applications.
```
Comment (greg-fer) [inline include/event_manager.rst:279]:
```suggestion
The Event Manager provides additional features that could be helpful when debugging event-based applications.
```

File: include/event_manager.rst [hunk 3]
```diff
@@ -307,8 +328,15 @@ The following code example shows how to define the event profiling information s
 .. note::
 	By default, all Event Manager events that are defined with an :c:struct:`event_info` argument are profiled.
 
+.. note::
+	:c:struct:`sample_event_info` is defined within :c:macro:`EVENT_INFO_DEFINE` macro.
+
+#. Use profiler scripts to profile the application. See :ref:`profiler` for more details.
```
Comment (greg-fer) [inline include/event_manager.rst:334]: Move the sentence to a new line.

**Review by greg-fer — CHANGES_REQUESTED** [review]
Some preliminary notes.

File: include/event_manager.rst (hunk 1 above)
Comment (emob-nordic) [reply include/event_manager.rst:197]: Won't do. Discussed offline.

**Comment by emob-nordic** [comment]
Review comments from @greg-fer applied.

**Review by greg-fer — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Comment by pdunaj** [comment]
It would be great to have this cleanup in.

**Pull request merged**